### PR TITLE
3628 timebox separate user creation from workspace creation

### LINKED ISSUE #3628: [Timebox] Separate user creation from workspace creation
As of today, when someone signs up we create the workspace along with the user in the signup method of the Auth service.
This clutters the database with a lot of unused schemas. 

Instead we should only create the workspace upon clicking on the workspace creation form if it's possible.

The issue is with billing on our cloud version... Currently the "plan required" screen is displayed based on the workspace status so that would break. I feel like we have to add a flag like `hasSetupBilling` or something like this on the `user` table to restrict access.

### PR DESCRIPTION
**Pull request by martmull**
This PR should be deployed after #3883
- Split workspace metadata and schema creation from user creation in signUp
- Create Workspace Schema and metadata in /create/workspace Form
- Add a defaultAvatarUrl column to core.user to store avatar on signUp to provide it to workspaceMember
- Add a new recoil value to indicate whether or not the workspaceSchema is created

### CODE REVIEW

File: packages/twenty-front/src/modules/auth/hooks/useAuth.ts [hunk 1]
```diff
@@ -89,13 +89,18 @@ export const useAuth = () => {
...
+        workspaceMember = {
+          ...user.workspaceMember,
+          colorScheme: user.workspaceMember?.colorScheme as ColorScheme,
+        };
+      }
       const workspace = user.defaultWorkspace ?? null;
       setCurrentUser(user);
-      setCurrentWorkspaceMember(workspaceMember);
+      if (workspaceMember) {
+        setCurrentWorkspaceMember(workspaceMember);
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/auth/hooks/useAuth.ts:102]: Could be placed above in `if(user.workspaceMember)` ?

File: packages/twenty-front/src/modules/auth/sign-in-up/hooks/useSignInUp.tsx [hunk 2]
```diff
@@ -143,16 +143,15 @@ export const useSignInUp = () => {
           billing?.isBillingEnabled &&
           currentWorkspace.subscriptionStatus !== 'active'
         ) {
-          navigate('/plan-required');
+          navigate(AppPath.PlanRequired);
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/auth/sign-in-up/hooks/useSignInUp.tsx:146]: Nice

**Review by lucasbordeau — APPROVED** [review]

File: packages/twenty-front/src/modules/auth/sign-in-up/hooks/useSignInUp.tsx (hunk 2 above)
Comment (martmull) [reply packages/twenty-front/src/modules/auth/sign-in-up/hooks/useSignInUp.tsx:146]: je l'ai fait partout dans l'app

**Review by charlesBochet — APPROVED** [review]
LGTM

**Pull request merged**